packages/datagateway-search/src/search/datePicker.component.tsx

```diff
@@ -41,6 +48,47 @@ interface DatePickerStoreProps {
 
 type DatePickerCombinedProps = DatePickerProps & DatePickerStoreProps;
 
+const CustomTextField: React.FC<TextFieldProps> = (
+  props: JSX.IntrinsicAttributes & TextFieldProps
+) => {
+  const invalidDateRange = props.inputProps?.invalidDateRange;
+  const [t] = props.inputProps?.t;
+  const sideLayout = props.inputProps?.sideLayout;
+
+  // remove the custom props so they are not passed to DOM
+  delete props.inputProps?.invalidDateRange;
+  delete props.inputProps?.t;
+  delete props.inputProps?.sideLayout;
```

Comment:
You could achieve all this on one line via object destructuring.

```
const { invalidDateRange, t, sideLayout, ...inputProps} = props.inputProps ?? {};
```

Then, you can change `inputProps` to just be `inputProps={inputProps}`